test/e2e/apimachinery/coverage.go

```diff
@@ -0,0 +1,389 @@
+
+	resourcesFile := generated.ReadOrDie(filepath.Join("test/e2e/testing-manifests/apiresource", "resources.csv"))
+
+	reader := csv.NewReader(bufio.NewReader(bytes.NewReader(resourcesFile)))
+	for {
+		record, err := reader.Read()
+		if err == io.EOF {
+			break
+		} else if err != nil {
+			return nil, fmt.Errorf("failed to read api resources file: %v", err)
+		}
+
+		err = parseResourceRecord(tables, record)
+		if err != nil {
+			return nil, fmt.Errorf("failed to parse resource record %v: %v", record, err)
+		}
+	}
+
+	return tables, nil
+}
+
+func parseResourceRecord(rMap resourceMap, record []string) error {
+	// Resource record should be in format of: group,version,name,namespaced,verb
+	if len(record) != 5 {
+		return fmt.Errorf("unexpected resource record length: %v, want: 5, got: %d", record, len(record))
+	}
+
+	namespaced, err := strconv.ParseBool(record[3])
+	if err != nil {
+		return fmt.Errorf("failed to parse resource (%v) namespaced property: %v", record, err)
+	}
+
+	if !strings.Contains(record[2], "/") {
+		// If record is a resource
+		m := resourceMeta{
+			group:   record[0],
+			version: record[1],
+			name:    record[2],
+		}
+		if rMap[m] == nil {
+			rMap[m] = new(resource)
+			*rMap[m] = resource{
+				group:        record[0],
+				version:      record[1],
+				name:         record[2],
+				namespaced:   namespaced,
+				verbs:        []string{record[4]},
+				subresources: resourceMap{},
+			}
+		} else {
+			rMap[m].verbs = append(rMap[m].verbs, record[4])
+		}
+	} else {
+		// If record is a subresource
+		parent := resourceMeta{
+			group:   record[0],
+			version: record[1],
+			name:    strings.Split(record[2], "/")[0],
+		}
+		// Enforce preparing resource before subresource in test data
+		if rMap[parent] == nil {
+			return fmt.Errorf("parent resource not found for subresource: %v", record)
+		}
+
+		m := resourceMeta{
+			group:   record[0],
+			version: record[1],
+			name:    record[2],
+		}
+		if rMap[parent].subresources[m] == nil {
+			rMap[parent].subresources[m] = new(resource)
+			*rMap[parent].subresources[m] = resource{
+				group:        record[0],
+				version:      record[1],
+				name:         record[2],
+				namespaced:   namespaced,
+				verbs:        []string{record[4]},
+				subresources: resourceMap{},
+			}
+		} else {
+			rMap[parent].subresources[m].verbs = append(rMap[parent].subresources[m].verbs, record[4])
+		}
+	}
+
+	return nil
+}
+
+func hasVerb(verbs []string, verb string) bool {
+	for _, v := range verbs {
+		if v == verb {
+			return true
+		}
+	}
+	return false
+}
+
+func (r *resource) dumpResourceYAML() *unstructuredv1.Unstructured {
+	unstruct := &unstructuredv1.Unstructured{}
+	if !hasVerb(r.verbs, "create") {
+		return unstruct
+	}
+
+	group := r.group
+	if group == "" {
+		group = "core"
+	}
+	yamlFile := generated.ReadOrDie(filepath.Join("test/e2e/testing-manifests/apiresource/yamlfiles/", fmt.Sprintf("%s/%s/%s", group, r.version, r.name)+".yaml"))
```

Comment:
this breaks on Windows with `/`, I would expect

Reply:
This "filename" is used as gobindata index, which gets converted into slash-style path in the end https://github.com/kubernetes/kubernetes/blob/e6b6e5c4b4f501daa61ce0554b36931104565cab/vendor/github.com/jteeuwen/go-bindata/toc.go#L165-L167

Added `filepath.FromSlash()` to make it more readable for Windows